std/Lambda.hx

```diff
@@ -75,7 +75,7 @@ class Lambda {
 		The order of elements is preserved.
 		If `f` is null, the result is unspecified.
 	**/
-	public static inline function mapi<A, B>(it:Iterable<A>, f:Int->A->B):Array<B> {
+	public static inline function mapi<A, B>(it:Iterable<A>, f:(index:A, item:B) -> B):Array<B> {
```

Comment:
`index:Int`

Reply:
also `item:A`

Reply:
And `item:A`.

Reply:
Also `item:A`

Reply:
or should that be `public static inline function mapi<A:Int, B>(it:Iterable<A>, f:(index:A, item:B) -> B):Array<B> {`

Reply:
No, it shouldn't. It's a map converting an iterable of `A`'s to an array of `B`'s. But since it is a `mapi` it also gives the index to the function.

Reply:
what? no. it iterates over a collection of `A` providing integer index as to the callback along with the element.

Reply:
Uh yeah you are right, I actually mean to add a some kind of `C:Int` but since you always just get an int, that doesn't make sense.

Reply:
You forgot `item:A` ;)